src/utils/optionsdict.cc

```diff
@@ -322,4 +322,17 @@ void OptionsDict::AddSubdictFromString(const std::string& str) {
   parser.ParseMain(this);
 }
 
+void OptionsDict::CheckAllOptionsRead(
+    const std::string& path_from_parent) const {
+  std::string s = path_from_parent.empty() ? "" : path_from_parent + '.';
```

Comment:
If you would add `'.'` at the function call at line 334, that check would not be needed at all.

Reply:
I would need to add the dot in factory.cc as well. Moreover, the idea is to be able to call this with an empty string, I'm not convinced `backend.option` is the best format for this (although this is the way I have it now).